### PR TITLE
Reveal old ability with Entrainment and Role Play

### PR DESCRIPTION
**Pull request by QuiteQuiet**
Actually how it works in game: https://youtu.be/h8aB0f03-do?t=3m15s

Appears to solve everything #2790 points out related toEntrainment.

### CODE REVIEW

**Comment by urkerab** [comment]
I see Simple Beam and Worry Seed already do this. Does anyone know what Role Play does on cart?

**Comment by QuiteQuiet** [comment]
Role Play also broadcast what the previous ability was before changing to the new one (https://youtu.be/DWdlHe3H2QI?t=3m40s). That's not what PS is doing currently, though.

**Comment by urkerab** [comment]
Well don't let me stop you from applying the same fix to Role Play too.

**Comment by Zarel** [comment]
So... do you want to commit the Role Play change to this pullreq before I merge?

**Comment by QuiteQuiet** [comment]
Added Role Play.

It might be worth considering to send `-endability` in the same place as I did for Role Play for the other moves too, since otherwise the message order is "activate new ability" -> "lose old ability" -> "get new ability" (pressure is a good example where this order happens).

**Comment by Zarel** [comment]
Well, `setAbility` can fail, right? Maybe the right place to do it is within `setAbility`.

**Comment by QuiteQuiet** [comment]
Something like this? To not get messages like Drought ending when Groudon-Primal is sent out there has to be some more information passed along. This is not a very stable solution for future moves though (or other things that end abilities like other abilities I guess).

**Comment by Zarel** [comment]
Yes, that's what I was thinking. You can read effect from `this.effect` instead of needing to pass it explicitly, though.

**Comment by QuiteQuiet** [comment]
`this.effect` appears to be undefined in `setAbility`, unless there's a way to set it before calling `setAbility`?

**Comment by Zarel** [comment]
Oh, it's `this.battle.effect`

**Comment by QuiteQuiet** [comment]
Something like this then.

File: battle-engine.js [hunk 1]
```diff
@@ -1062,6 +1062,9 @@ class BattlePokemon {
 			if (oldAbility in {multitype:1, stancechange:1}) return false;
 		}
 		this.battle.singleEvent('End', this.battle.getAbility(oldAbility), this.abilityData, this, source, effect);
+		if (this.battle.effect && this.battle.effect.id in {entrainment:1, roleplay:1, simplebeam:1, worryseed:1}) {
```
Comment (Zarel) [inline battle-engine.js]: I don't think hardcoding is the best idea, but I don't exactly have a better idea either. :(

**Pull request merged**